moralization/transformers_data_handler.py

```diff
@@ -94,3 +101,67 @@ def init_tokenizer(self, model_name="bert-base-cased", kwargs=None):
             raise ValueError(
                 "Please use a different model that provices a fast tokenizer"
             )
+
+    def tokenize(self, wordlist: list = None):
+        if wordlist is None:
+            wordlist = self.token_list
+        wordlist = self._check_is_nested(wordlist)
+        self.inputs = self.tokenizer(
+            wordlist, truncation=True, is_split_into_words=True
+        )
+
+    def _check_is_nested(self, list_to_check: list) -> list:
+        # make sure that data is a nested list,
+        # otherwise add a layer
+        # do we need this?
+        # maybe enough to check if list_to_check[0] is a list?
+        # does it cost us to iterate over all the data?
+        list_to_check = (
+            [list_to_check] if not isinstance(list_to_check[0], list) else list_to_check
+        )
+        list_to_check = [[i] if not isinstance(i, list) else i for i in list_to_check]
+        return list_to_check
+
+    def _align_labels_with_tokens(self, labels: list, word_ids: list):
+        """Helper method to expand the label list so that it matches the new tokens."""
+        # beginning of a span needs a label of 2
+        # inside a span label of 1
+        # punctuation is ignored in the calculation of metrics: set to -100
+        new_labels = [
+            -100 if word_id is None else labels[word_id] for word_id in word_ids
+        ]
+        # if the beginning of a span has been split into two tokens,
+        # make sure that the label "2" only appears once
+        # seems to me we need to use enumerate
+        new_labels = [
+            1 if label == 2 and new_labels[i - 1] == 2 and i >= 1 else label
```

Comment:
I think the check that i>0 should come before indexing `i-1`
```suggestion
            1 if label == 2 and i >= 1 and new_labels[i - 1] == 2 else label
```